libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -379,9 +379,13 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> {
                 case AdaptiveEvents.recognizeUtterance:
                     if (activity.type == ActivityTypes.Message) {
                         // Recognize utterance
-                        const recognized = await this.onRecognize(actionContext, activity);
+                        const recognizedResult = await this.onRecognize(actionContext, activity);
                         // TODO figure out way to not use turn state to pass this value back to caller.
-                        actionContext.state.setValue(TurnPath.recognized, recognized);
+                        actionContext.state.setValue(TurnPath.recognized, recognizedResult);
+                        const { intent, score } = getTopScoringIntent(recognizedResult);
+                        actionContext.state.setValue(TurnPath.topIntent, intent);
+                        actionContext.state.setValue(TurnPath.topScore, score);
+                        actionContext.state.setValue(DialogPath.lastEvent, intent);
```

Comment:
>lastEvent [](start = 64, length = 9)

This is incorrect.  An intent is not the event.  Did you run tests and this passed?

Reply:
I think it should be `DialogPath.lastIntent`. @songcs
```suggestion
                        actionContext.state.setValue(DialogPath.lastIntent, intent);
```

Reply:
Sorry, this bug has now been fixed. It should be DialogPath.lastIntent.